packages/image-block/src/hooks/useImageContainer.tsx

```diff
@@ -0,0 +1,40 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { debounce } from '@frontify/fondue';
+import { useEffect, useRef, useState } from 'react';
+import { roundToNextHundred } from '../helpers';
+
+export const useImageContainer = () => {
+    const containerRef = useRef<HTMLElement | null>(null);
+    const [containerWidth, setContainerWidth] = useState<number | undefined>(undefined);
+
+    useEffect(() => {
+        if (!containerRef.current) {
+            return;
+        }
+
+        const containerObserver = new ResizeObserver(
+            debounce((entries) => {
+                const container = entries[0];
+                const newContainerWidth = roundToNextHundred(container.contentRect.width);
+                const oldContainerWidth = roundToNextHundred(containerWidth ?? 0);
```

Comment:
containerWidth is always a value of 100 so I think this is unneccessary to wrap in the fn here